lab_5_scrapper/scrapper.py

```diff
@@ -127,11 +235,28 @@ def _extract_url(self, article_bs: BeautifulSoup) -> str:
         Returns:
             str: Url from HTML
         """
+        url = ''
+        for div in article_bs.find_all('div', {'class': 'post-details'}):
+            for urls in div.select('a'):
```

Comment:
this method should extract just one url from one beautifulSoup element given, so why you try to iterate over a list of urls? it seems to me that you just save the last url and rewrite previous urls every iteration